### PR TITLE
Fix node metadata fetch in stackdriver filter

### PR DESCRIPTION
**Pull request by bianpengyuan**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by howardjohn** [comment]
you scared me with the 40k line change... I think you accidentally included a log

**Comment by bianpengyuan** [comment]
Yes going to remove it when finalize this pr.

**Comment by mandarjog** [comment]
This contains a few unrelated changes.
1.  Add x-google-project to edges and logging by using `GrpcCallHandler`
2. Add x-google-project to metrics by using an interceptor.
3. The latency `ms` fix from nanos to milis.
4. fixing STSPort fetching from metadata
5. fetching project_id from node metadata

For milis and STS port a test is needed. We can add it in master / 1.5 if adding it here takes too long.

**Comment by bianpengyuan** [comment]
@mandarjog @kyessenov @gargnupur Please help on reviewing this PR when you get some time, thanks!

**Comment by bianpengyuan** [comment]
Based on chat with Mandar, will add a test for the duration conversion in stackdriver.

File: extensions/stackdriver/edges/mesh_edges_service_client.cc [hunk 1]
```diff
@@ -98,9 +126,11 @@ MeshEdgesServiceClientImpl::MeshEdgesServiceClientImpl(
 
 void MeshEdgesServiceClientImpl::reportTrafficAssertions(
     const ReportTrafficAssertionsRequest& request) const {
-  context_->grpcSimpleCall(
+  context_->grpcCallHandler(
       grpc_service_, kMeshEdgesService, kReportTrafficAssertions, request,
-      kDefaultTimeoutMillisecond, success_callback_, failure_callback_);
+      kDefaultTimeoutMillisecond,
+      std::unique_ptr<GrpcCallHandlerBase>(
```
Comment (kyessenov) [inline extensions/stackdriver/edges/mesh_edges_service_client.cc]: make_unique?

File: extensions/stackdriver/metric/registry.cc [hunk 2]
```diff
@@ -50,11 +94,23 @@ StackdriverOptions getStackdriverOptions(
...
+      file_string << file.rdbuf();
+      ssl_creds_options.pem_root_certs = file_string.str();
+    }
     auto channel_creds = grpc::SslCredentials(ssl_creds_options);
-    auto channel = ::grpc::CreateChannel(
+    grpc::ChannelArguments args;
+    std::vector<
+        std::unique_ptr<grpc::experimental::ClientInterceptorFactoryInterface>>
+        creators;
+    creators.push_back(std::unique_ptr<GoogleUserProjHeaderInterceptorFactory>(
```
Comment (gargnupur) [inline extensions/stackdriver/metric/registry.cc]: nit: make_unique here too

File: extensions/stackdriver/metric/registry.cc [hunk 3]
```diff
@@ -24,6 +27,47 @@ namespace Extensions {
...
+  GoogleUserProjHeaderInterceptorFactory(const std::string& project_id)
+      : project_id_(project_id) {}
+
+  virtual grpc::experimental::Interceptor* CreateClientInterceptor(
+      grpc::experimental::ClientRpcInfo*) override {
+    return new GoogleUserProjHeaderInterceptor(project_id_);
+  }
+
+ private:
+  std::string project_id_;
```
Comment (gargnupur) [inline extensions/stackdriver/metric/registry.cc]: can make this const std::string& too?

**Comment by bianpengyuan** [comment]
@kyessenov @mandarjog @gargnupur ptal, thanks!. John will merge this once this is lgtm-ed.

File: extensions/stackdriver/metric/record.cc [hunk 4]
```diff
@@ -28,7 +28,7 @@ namespace Metric {
 void record(bool is_outbound, const ::wasm::common::NodeInfo &local_node_info,
             const ::wasm::common::NodeInfo &peer_node_info,
             const ::Wasm::Common::RequestInfo &request_info) {
-  double latency_ms = request_info.duration / absl::Nanoseconds(1) / 1000.0;
+  double latency_ms = request_info.duration / absl::Milliseconds(1);
```
Comment (kyessenov) [inline extensions/stackdriver/metric/record.cc]: This is a bug that is not about stackdriver filter. or STS Please submit this to master.

File: test/envoye2e/stats/stats_xds_test.go [hunk 5]
```diff
@@ -0,0 +1,314 @@
+// Copyright 2019 Istio Authors
```
Comment (kyessenov) [inline test/envoye2e/stats/stats_xds_test.go]: bad merge?

**Pull request merged**